### DISCUSSION TITLE
How do I ignore all Python files in project root?

### DISCUSSION THREAD

**Discussion by elliot-100** [astral-sh/ruff#13730] [Q&A] [answered]
I'm sure this has a simple answer, but I haven't found it or worked it out.

I know how to include or exclude Python files in given directories, but is there a way of

- Excluding files in project root
and/or
- Including all files in all subfolders?

**Comment by MichaReiser** [comment, marked as answer]
> Excluding files in project root and/or

I'm not sure I understand what you're trying to do. Do you want to exclude specific files or all files in the root directory? 

I think the following allows to only include python files that are in a subdirectory but not files that are in the root directory. 

```toml
include = ["*/**/*.py"]
```

**Comment by elliot-100** [comment]
Sorry I wasn't clear - I meant exclude all files in the root directory. My either/or was intended to be  about whether it could be implemented with an include or exclude pattern or both. Will try what you suggest - thanks.
